### PR TITLE
Notification: Implement notification option for channels

### PR DESCRIPTION
**Pull request by BakasuraRCE**
Hi @TingPing,

I implemented the flow of notifications for channels as we accorded:

Flow:

1. You can control the notifications in a global environment
2. chanopts per each channel/pm/network MUST be followed overloading any other setting

Related: https://github.com/hexchat/hexchat/issues/876 https://github.com/hexchat/hexchat/issues/1172

Regards.

### CODE REVIEW

File: src/common/plugin.c [hunk 1]
```diff
@@ -1582,48 +1614,38 @@ hexchat_list_int (hexchat_plugin *ph, hexchat_list *xlist, const char *name)
...
-			tmp <<= 1;
-			tmp |= ((struct session *)data)->server->end_of_motd;/* 3 */
-			tmp <<= 1;
-			tmp |= ((struct session *)data)->server->is_away;    /* 2 */
-			tmp <<= 1;
-			tmp |= ((struct session *)data)->server->connecting; /* 1 */ 
-			tmp <<= 1;
-			tmp |= ((struct session *)data)->server->connected;  /* 0 */
-			return tmp;
+            channel_flags[0] = ((struct session *)data)->server->connected;
```
Comment (TingPing) [inline src/common/plugin.c:1617]: Nice cleanup, can you keep it in a separate commit.

File: src/fe-gtk/plugin-notification.c [hunk 2]
```diff
@@ -116,21 +119,50 @@ incoming_hilight_cb (char *word[], gpointer userdata)
...
 static int
 incoming_priv_cb (char *word[], gpointer userdata)
 {
 	int priv;
-
-	if (hexchat_get_prefs (ph, "input_balloon_priv", NULL, &priv) == 3 && priv && should_alert ())
+	int flags;
+    int alert = 0;
+
+    flags = hexchat_list_int(ph, NULL, "flags");
```
Comment (TingPing) [inline src/fe-gtk/plugin-notification.c:152]: Need spaces after function throughout commit.

File: src/common/hexchat.h [hunk 3]
```diff
@@ -363,6 +363,7 @@ typedef struct session
 {
 	/* Per-Channel Alerts */
 	/* use a byte, because we need a pointer to each element */
+    guint8 alert_balloon;
```
Comment (TingPing) [inline src/common/hexchat.h:366]: Throughout commit you mix tabs and spaces. This project uses tabs.

File: src/common/plugin.c [hunk 4]
```diff
@@ -1521,7 +1549,11 @@ hexchat_list_int (hexchat_plugin *ph, hexchat_list *xlist, const char *name)
 {
 	guint32 hash = str_hash (name);
 	gpointer data = ph->context;
-	int tmp = 0;
+
+    int channel_flag;
+	int channel_flags[CHANNEL_FLAG_COUNT];
+	int channel_flags_flag = 0;
```
Comment (TingPing) [inline src/common/plugin.c:1555]: These names are a bit easy to confuse, no better suggestion off the top of my head.

**Comment by BakasuraRCE** [comment]
I reopen here: https://github.com/hexchat/hexchat/pull/2491

File: src/common/plugin.c (hunk 4 above)
Comment (BakasuraRCE) [reply src/common/plugin.c:1555]: I change to used
